client.go

```diff
@@ -148,6 +148,9 @@ func (c *Client) handleErrorResp(resp *http.Response) error {
 			HTTPStatusCode: resp.StatusCode,
 			Err:            err,
 		}
+		if errRes.Error != nil {
```

Comment:
The if condition above checks for `err != nil || errRes.Error == nil`, so it's possible that `errRes` is nil here and you'll dereference a nil pointer